### PR TITLE
Remove duplication

### PR DESCRIPTION
**Pull request by nightblade9**
Unfortunately, it includes your resting changes too. `/shrug`

### CODE REVIEW

File: furusiya/model/maps/area_map.py [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 
 class AreaMap:
+
+    # TODO: these should be the renderer's concern, not the map's.
```
Comment (NegativeScript) [inline furusiya/model/maps/area_map.py]: Or perhaps the map generator's?
Either way, one thing's for sure: they don't belong here.

File: furusiya/model/maps/area_map.py [hunk 2]
```diff
@@ -57,15 +58,31 @@ def place_on_random_ground(self, entity):
...
 
+# TODO: extract into its own file
+class MapTile:
+    def __init__(self, walkable=False):
         self.is_explored = False
+        self.convert_to_ground()
 
-        # by default, if a tile is blocked, it also blocks sight
-        self.block_sight = block_sight or not self.is_walkable
+    def convert_to_wall(self, character=AreaMap.WALL_CHARACTER, colour=AreaMap.LIGHT_WALL_COLOUR, dark_colour=AreaMap.DARK_WALL_COLOUR):
```
Comment (NegativeScript) [inline furusiya/model/maps/area_map.py]: I seriously don't like this using constants from another class. Perhaps they belong in here instead?

File: furusiya/model/maps/generators/dungeon_generator.py [hunk 3]
```diff
@@ -100,32 +99,16 @@ def _generate_rooms(self):
...
-        map_tile.dark_colour = color_dark_ground
-
-    # TODO: DRY with forest generator
-    def _convert_to_wall(self, map_tile):
-        map_tile.is_walkable = False
-        map_tile.block_sight = True
-        map_tile.character = WALL_CHARACTER
-        map_tile.colour = color_light_wall
-        map_tile.dark_colour = color_dark_wall
+            self._area_map.tiles[x][y].convert_to_ground()
```
Comment (NegativeScript) [inline furusiya/model/maps/generators/dungeon_generator.py]: This is good. Nice work

File: furusiya/model/maps/area_map.py (hunk 2 above)
Comment (nightblade9) [reply furusiya/model/maps/area_map.py]: I don't know. I think anything to do with display (character, colour) should be ripped out of this "model" class. A common pattern I've seen is MV (ModelView) classes, eg. a `TileView` class that has a `MapTile` and a character/colour. Let's think on this.

**Pull request merged**